test/parallel/test-transform-by.js

```diff
@@ -286,17 +263,17 @@ async function transformByThrowPriorToForAwait() {
     strictEqual(err.message, 'kaboom');
   }));
 
-  read.pipe(stream);
+  read.pipe(stream).resume();
 }
 
 Promise.all([
   transformBy(),
   transformByFuncReturnsObjectWithSymbolAsyncIterator(),
-  transformByObjReturnedWSymbolAsyncIteratorWithNonPromiseReturningNext(),
```

Comment:
@davidmarkclements: I don't think this test is needed. Returning a non promise is fine?